src/backend/utils/page-builder.js

```diff
@@ -0,0 +1,242 @@
+ *       subtitle: 'Latest comments from user all around the world',
+ *       columns: 6,
+ *       items: mappedComments
+ *     })
+ *     await this.addTextBox({
+ *       title: 'Simple textbox',
+ *       content: '<div> lorem ipsum contentum textum boxum</div>',
+ *       columns: 6
+ *     })
+ *     await this.addChart({
+ *       columns: 6,
+ *       config: {
+ *         name: 'example',
+ *         type: 'line',
+ *         data: {
+ *           labels: clients.map(client => client.month),
+ *           datasets: [
+ *             {
+ *               label: 'label example',
+ *               fill: true,
+ *               backgroundColor: 'orange',
+ *               borderColor: 'tomato',
+ *               data: clients.map(client => client.amount),
+ *             },
+ *           ],
+ *         },
+ *         options: {
+ *           title: {
+ *             fontSize: 20,
+ *             display: true,
+ *             text: 'Title'
+ *           },
+ *         },
+ *       }
+ *     })
+ *   }    
+ * }
+ */
+
+class PageBuilder {
+  /**
+   *
+   * @param  {AdminBro}     options.admin  current instance of AdminBro
+   * @param  {String[]}     _pageContent   array of html elements as String
+   * @param  {String}       _pageHeader    html element as String
+   * @param  {String}       title          page title
+   * @param  {Object}       charts         mapped object contains chart's settings
+   * @param  {Object}       colorTypes     specific colors for html blocks
+   */
+  constructor({ admin }) {
+    this._admin = admin
+    this._pageContent = []
+    this._pageHeader = ''
+    this.title = ''
+    this.charts = {}
+    this.colorTypes = {
+      warning: '#ff9f89',
+      danger: '#f0616f',
+      succes: '#21c197',
+      info: '#718af4',
+    }
+  }
+
+  /**
+   * Returns object with page settings
+   * @return {Object}
+   */
+  async render() {
+    await this.build()
+    return {
+      title: this.title,
+      header: this._pageHeader,
+      content: this.convertedPageContent(),
+      charts: this.charts
+    }
+  }
+
+  /**
+   * This method is responsible for building the page, should be overriden.
+   */
+  build() {
+    throw new Error('You have to overwrite this method')
+  }
+
+  /** Returns string of html content elements
+   * @return {String}
+   */
+  convertedPageContent() {
+    if (this._pageContent.length > 0) {
+      return this._pageContent.join('')
+    }
+    return null
+  }
+
+  /** Adjusts default dashboard content as a pageHeader
+   * 
+   */
+  async addDefaultDashboard() {
+    this._pageHeader =  await new Renderer('pages/defaultDashboard').render()
+  }
+
+  /** Adjusts overview content as a pageHeader
+   * 
+   */
+  async addOverview(title = '', subtitle = '') {
+    this._pageHeader = await  new Renderer('partials/overview', { title, subtitle }).render()
+  }
+
+  /** Adds canvas chart to the page content
+   * 
+   */
+  async addChart(options) {
```

Comment:
I would add info what could options be -> to be more precised: what could be options.config. (maybe link to chart js)